snapshots/gcp/2023-12-12/global_carbon_budget.py

```diff
@@ -0,0 +1,67 @@
+
+DESCRIPTION = """The Global Carbon Budget was founded by the Global Carbon Project (GCP) international science team to track the trends in global carbon emissions and sinks and is a key measure of progress towards the goals of the Paris Agreement. It's widely recognized as the most comprehensive report of its kind.
+
+The GCP has been publishing estimates of global and national fossil CO2 emissions since 2001. In the first instance these were simple re-publications of data from another source, but over subsequent years refinements have been made in response to feedback and identification of inaccuracies."""
+
+
+@click.command()
+@click.option(
+    "--upload/--skip-upload",
+    default=True,
+    type=bool,
+    help="Upload dataset to Snapshot",
+)
+def main(upload: bool) -> None:
+    # Create a new snapshot for each dataset.
+    for data_file in DATA_FILES:
+        snap = Snapshot(f"gcp/{SNAPSHOT_VERSION}/{data_file}")
+
+        # Replace the full citation and description in the metadata.
+        snap.metadata.origin.attribution = ATTRIBUTION  # type: ignore
+        snap.metadata.origin.citation_full = CITATION_FULL  # type: ignore
+        snap.metadata.origin.description = DESCRIPTION  # type: ignore
```

Comment:
Whatever comment I write to the snapshot `dvc` file will be overwritten when the `.py` script is executed (given that the dictionary of metadata is converted into a yaml file, without respecting comments).

Reply:
Yes, my point was that you don't need to write all that long text in the DVC, and could just write a short sentence saying that it will be replaced during runtime.

Reply:
Currently, as far as I know, that's not possible: The content of the `dvc` file needs to be the "final one", i.e. the one you get after running the .py file. There's no under-the-hood filling processing of dvc files.

Reply:
Ah, I see! So what I am seeing in the `dvc` files is actually kinda 'read-only'. I guess that's a good way of looking at it: `yaml` files should be read-only, only editable via (i) wizard or (ii) code atm.

Reply:
Incidentally, I'm working on replacing some of the [DVC functionality](https://github.com/owid/etl/pull/2058). That patch would make it possible to add comments to `*.dvc` files (and give us more control in general). I'm not yet sure whether I want to merge it, but it might be worth discussing.